src/ralph/attachments/views.py

```diff
@@ -84,3 +75,24 @@ def formset_invalid(self, formset):
         return self.render_to_response(
             self.get_context_data(formset=formset)
         )
+
+
+class ServeAttachment(View):
+
+    def get(self, *args, **kwargs):
+        """
+        All attachments are serving by this view because we need full
+        control (e.g., permissions, rename).
+        """
+        # TODO: respect permissions
+        obj = get_object_or_404(
+            Attachment,
+            id=self.kwargs['id'],
+            original_filename=self.kwargs['filename']
+        )
+        fd = open(obj.file.path, 'rb')
```

Comment:
you're not closing file descriptor. use context manager here and wrap it in try-except for open-related exceptions (raise 4XX)